### PR TITLE
refactor validations for command component

### PR DESCRIPTION
**Pull request by novaturient95**
refactor to do env validation for command component alone

### CODE REVIEW

File: scripts/release/asset_publish.py [hunk 1]
```diff
@@ -138,6 +138,57 @@ def prepare_model(model_config: assets.ModelConfig, spec_file_path: Path, model_
     return can_publish_model
 
 
+def validate_command_component(component: Component, spec_path: Path, final_version: str) -> bool:
```
Comment (elesel) [inline scripts/release/asset_publish.py:141]: Consider a more descriptive name, to make it clear that this function changes the spec files:
```suggestion
def validate_update_command_component(component: Component, spec_path: Path, final_version: str) -> bool:
```

File: scripts/release/asset_publish.py [hunk 2]
```diff
@@ -296,51 +347,10 @@ def _str2bool(v: str) -> bool:
...
-                    failure_list.append(asset)
-                    continue
-
-                component.environment = env.id
-                if not update_spec(component, asset.spec_with_path):
-                    logger.print(f"Component update failed for asset spec path: {asset.spec_with_path}")
-                    failure_list.append(asset)
-                    continue
+                if component.type == "command":
+                    if not validate_command_component(component, asset.spec_with_path, final_version):
```
Comment (elesel) [inline scripts/release/asset_publish.py:351]:
```suggestion
                    if not validate_update_command_component(component, asset.spec_with_path, final_version):
```

Comment (novaturient95) [reply scripts/release/asset_publish.py]: Done

File: scripts/release/asset_publish.py (hunk 1 above)
Comment (novaturient95) [reply scripts/release/asset_publish.py]: Done

File: scripts/release/asset_publish.py [hunk 3]
```diff
@@ -138,6 +138,64 @@ def prepare_model(model_config: assets.ModelConfig, spec_file_path: Path, model_
...
+        return False
+
+    # both ws and registry env follows the same grouping
+    env_name, env_version, env_label = match.group(1), match.group(2), match.group(3)
+    logger.print(f"Env name: {env_name}, version: {env_version}, label: {env_label}")
+
+    if env_label:
+        # TODO: Add fetching env from label
+        # https://github.com/Azure/azureml-assets/issues/415
+        logger.print("Unexpected !!! Registering a component with env label is not yet supported.")
```
Comment (vizhur) [inline scripts/release/asset_publish.py:177]: I would remove `yet` to not set any expectations

Comment (novaturient95) [reply scripts/release/asset_publish.py:177]: :D, removed.

**Pull request merged**